src/models/AppController.js

```diff
@@ -0,0 +1,86 @@
+import _ from 'lodash';
+import { Logger } from 'zos-lib';
+import { parseJsonIfExists, writeJson } from "zos-lib/lib/utils/FileSystem";
+import { ContractsProvider, AppManagerProvider, AppManagerDeployer } from "zos-lib";
+import StdlibProvider from './stdlib/StdlibProvider';
+import StdlibDeployer from './stdlib/StdlibDeployer';
+import Stdlib from './stdlib/Stdlib';
+import NetworkAppController from './NetworkAppController';
+
+const log = new Logger('AppController');
+
+const DEFAULT_VERSION = '0.1.0';
+
+export default class AppController {
+  constructor(packageFileName) {
+    this.packageFileName = packageFileName || 'package.zos.json';
+  }
+
+  onNetwork(network, txParams, networkFileName) {
+    return new NetworkAppController(this, network, txParams, networkFileName);
+  }
+
+  init(name, version) {
+    if (this.package.name) {
+      throw new Error(`Cannot initialize already initialized package ${this.package.name}`)
```

Comment:
is it the same as doing `throw 'bla'`?

Reply:
I understand that throwing an instance Error is cleaner, but I cannot find any reliable reference now.